src/store/ChartState.js

```diff
@@ -113,7 +113,7 @@ class ChartState {
 
         if (chartType !== this.chartType && this.context) {
             if (chartType === 'table') this.prevChartType = this.chartTypeStore.type.id;
-            this.setChartType(chartType);
+            this.setChartChartType(chartType);
```

Comment:
@shayan-fs shouldn't it be `this.setChartType`?
```suggestion
            this.setChartType(chartType);
```

Reply:
this is use to be `setChartType` then I changed it to `setChartChartType` and add `setChartType` to the state. `setChartChartType` change the chart type, `setChartType` change the state store chart-type.

Reply:
@shayan-fs but the name is like a typo and make some confusing it'd be better to choose a better name.

Reply:
@morteza-binary two option, 1) rename `setChartChartType` to `setCIQChartType`. or rename `setChartType` to `setStateChartType` and  `setChartChartType` to `setChartType`. should notice that by this rule all state function which has this issue renamed so that we have a constant name structure for all methods